### DISCUSSION TITLE
I'm not able to setup the Slack integration

### DISCUSSION THREAD

**Discussion by daladim** [outline/outline#3169] [Self Hosting] [answered]
Hello.

I succeeded in setting up a self-hosted instance of Outline (I have fixes to suggest btw, see below).
I can correctly login with a Slack SSO, and I'm now trying to get the Slack integration working. I went to `Settings > Integrations > Slack` and clicked the `Add to Slack` button.

This redirects me onto `https://my-company.slack.com/oauth?client_id=1000000000000000&scope=commands+links%3Aread+links%3Awrite&user_scope=&redirect_uri=https%3A%2F%2Foutline.my-company.com%3A8080%2Fauth%2Fslack.commands&state=...&granular_bot_scope=0&single_channel=0&install_redirect=&tracked=1&team=` , which has the following error message
> Something went wrong when authorizing My-Outline-App
>
> Error details
> redirect_uri did not match any configured URIs. Passed URI: https://outline.my-company.com:8080/auth/slack.commands

The `auth/slack.commands` is nowhere to be found on your documentation. Am I missing something?

(I have set `SLACK_VERIFICATION_TOKEN` and `SLACK_APP_ID` in my `docker.env` if that matters).

Thanks!

Btw, here are things that I have changed from the sample docker-compose.yml file from your documentation so that Outline started working. I'm not a Docker expert, but you might have a look, maybe that's indeed something that should be fixed:
* the ports for the Redis container should be `"6479:6379"`
* the ones for the Postgres container should be `"5532:5432"`
* the http-portal configuration should probably be fed with a `environment: DOMAINS` and `environment: STAGE: 'production'`
* in the env.sample, I had to add `POSTGRES_USER=user`, `POSTGRES_PASSWORD=pass` and `POSTGRES_DB=outline`

**Comment by tommoor** [comment, marked as answer]
You should add the url to the list of OAuth redirect urls in the Slack app configuration

**Reply by daladim** [reply]
I'm very sorry I had not answered before.
I finally managed to resolve this issue (probably thanks to your message?) but I'm afraid I cannot remember exactly what fixed it.
Thank your very much for your answer anyway (and for all the work you've done on Outline!)

**Comment by macgate** [comment]
Hello, i have the next step problem.
After adding the url in my OAuth redirect urls list in Slack app.
I have the good authorization page from slack, after submitting it enter in a server loop.
Slack load the url `https://docs.****.com/auth/slack.commands?code=2***65267456.4672026371412.ce38e78b10f3b115a65980d91f40584bef6852bf464b4c4441f618ef2f33f397&state=c106fc1b-8a56-4db6-887e-5f76094c556*` who charge a page which have a refresh from that same URL.
```
<html>
<head>
	<meta http-equiv="refresh"
		content="0;URL='https://docs.****.com/auth/slack.commands?code=2***65267456.4672026371412.ce38e78b10f3b115a65980d91f40584bef6852bf464b4c4441f618ef2f33f397&state=c106fc1b-8a56-4db6-887e-5f76094c556*'" />
</head>
```
In result, it's an infinite loop and i can't connect me to my slack app.
Do you think a reason ?
Thanks
I have configured All token of slack correctly in my .env file : 
`SLACK_CLIENT_ID=, SLACK_CLIENT_SECRET=, SLACK_KEY=, SLACK_SECRET=, SLACK_VERIFICATION_TOKEN=, SLACK_APP_ID=, SLACK_MESSAGE_ACTIONS=true`

**Reply by tommoor** [reply]
Strange but it would indicate that you do not have a logged in session on the root docs. domain

**Reply by macgate** [reply]
But i have because for accessing the slack button to start the process, i have to be logged.
Also, i'm the administrator of my docs. domain.
For your information, my login access is not from Slack, but from Google.
End in my BDD in the table `authentication_providers` i have only the provider `google`.
Is the provider `slack` is necessary for doing the connection ? If so, how to add it ?

**Reply by tommoor** [reply]
It is not necessary to have an authentication_provider setup for the Slack integration. I don't know why the redirect is happening I'm afraid and not in a position to debug your installation
